### ISSUE TITLE
Must the format of the input training set be 416x416

### ISSUE THREAD

**Issue by JCRONG96** [zzh8829/yolov3-tf2#178]
If I need to train my data in other formats, do I need to adjust my format first?

**Comment by Ludwig-Zh** [comment]
No:
" train_dataset = train_dataset.map(lambda x, y: (
        dataset.transform_images(x, FLAGS.size),
        dataset.transform_targets(y, anchors, anchor_masks, FLAGS.size)))"--train.py
Other size can be accepted.

**Comment by JCRONG96** [comment]
> 号：
> “train_dataset = train_dataset.map（拉姆达X，Y：（
> dataset.transform_images（X，FLAGS.size），
> dataset.transform_targets（Y，锚，anchor_masks，FLAGS.size）））” - train.py
> 其它尺寸可以接受。

do you think using "tf.image.resize_with_pad"  instead of  "tf.image.resize" is better? 
This is what I want to ask.
The documentation says： Resized images will be distorted if their original aspect ratio is not the same as size. To avoid distortions see tf.image.resize_with_pad.

**Comment by JCRONG96** [comment]
> No:
> " train_dataset = train_dataset.map(lambda x, y: (
> dataset.transform_images(x, FLAGS.size),
> dataset.transform_targets(y, anchors, anchor_masks, FLAGS.size)))"--train.py
> Other size can be accepted.

do you think using "tf.image.resize_with_pad" instead of "tf.image.resize" is better?
This is what I want to ask.
The documentation says： Resized images will be distorted if their original aspect ratio is not the same as size. To avoid distortions see tf.image.resize_with_pad.

**Comment by HotThoughts** [comment]
Hi, thanks for the question. I also have the same concern. One more question, do we need to update the position of bounding box accordingly if we resize the image with padding?

**Comment by JCRONG96** [comment]
> Hi, thanks for the question. I also have the same concern. One more question, do we need to update the position of bounding box accordingly if we resize the image with padding?

Using padding to adjust the image size will change the aspect ratio, but in the VOC2012.py file, the size of the bounding box in the xml file is normalized. If the image is filled, the proportion of the box will change.

**Comment by JCRONG96** [comment]
The following  is just my superficial understanding. Using tf.image.resize_with_pad to adjust the image size will change the aspect ratio, but in the VOC2012.py file, the size of the bounding box in the xml file is normalized. If the image is filled, the proportion of the box will change. Using the tf.image.resize, the proportion do not change with the distorted image.

**Comment by VeeranjaneyuluToka** [comment]
Is preserving aspect ratio of an image improves the model accuracy further?
